core/src/main/scala/scala/collection/immutable/TrieIterator.scala

```diff
@@ -39,7 +39,7 @@ private[collection] abstract class TrieIterator[+T](elems: Array[Iterable[T]]) e
   def initSubIter: Iterator[T]                      = null // to traverse collision nodes
 
   private[this] var depth                                     = initDepth
-  private[this] var arrayStack: Array[Array[Iterable[T @uV]]] = initArrayStack
+  private[this] val arrayStack: Array[Array[Iterable[T @uV]]] = initArrayStack
```

Comment:
This is @joroKr21 in action. Perhaps there is confidence wrt this code in the comment
```
// we know that `this` is not empty, since it had something on the arrayStack and arrayStack elements are always non-empty
```